optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -972,25 +1021,21 @@ public int getValueCount(Solution_ solution) {
      * @return {@code >= 0}
      */
     public long getProblemScale(Solution_ solution) {
-        return extractAllEntitiesStream(solution)
-                .mapToLong(entity -> findEntityDescriptorOrFail(entity.getClass()).getProblemScale(solution, entity))
-                .sum();
+        LongAdder adder = new LongAdder();
+        visitAllEntities(solution,
+                entity -> adder.add(findEntityDescriptorOrFail(entity.getClass()).getProblemScale(solution, entity)));
+        return adder.longValue();
     }
 
     public int countUninitializedVariables(Solution_ solution) {
-        long count = extractAllEntitiesStream(solution)
-                .mapToLong(entity -> findEntityDescriptorOrFail(entity.getClass()).countUninitializedVariables(entity))
-                .sum();
+        LongAdder adder = new LongAdder();
```

Comment:
@triceo You sure that `LongAdder` is the best option? It is designed for concurrent modification so there may be some unnecessary overhead.

Reply:
The reason for this class is that I need a final reference I can use inside the lambda. So `Long` will not do. I can use `AtomicLong`, but that has the same issue. Maybe I could use `long[1]`.

Reply:
Yeah, the reason is clear.

From `LongAdder`:
> This class is usually preferable to `AtomicLong` when multiple threads update a common sum [...]. Under low update contention, the two classes have similar characteristics. But under high contention, expected throughput of this class is significantly higher, at the expense of higher space consumption.

So unless you need frequent updates from many threads, `LongAdder` is probably space-inefficient and `AtomicLong` can be used instead. But `AtomicLong` does not have the `add()` method. It only has `getAndAdd()` or `addAndGet()`, which would work fine, but IMO wouldn't help with the confusion:
1. Why are we using a concurrency class for a single-threaded iteration?
2. Why do we call `addAndGet()` if the return value is ignored?

IMO, something like [MutableInt](https://github.com/apache/commons-lang/blob/master/src/main/java/org/apache/commons/lang3/mutable/MutableInt.java#L181) or [MutableLong](https://github.com/apache/commons-lang/blob/master/src/main/java/org/apache/commons/lang3/mutable/MutableLong.java#L181) would be best. If we don't want to drag `commons-lang3` into the `compile` scope, I'd be fine even with introducing a dedicated helper class that does exactly and only what's needed for this purpose. Same as you've introduced `Pair` and `MutablePair`. Pros: clarity and efficiency.

`long[1]` == top efficiency, poor clarity.